modin/experimental/cloud/rayscale.py

```diff
@@ -133,8 +133,32 @@ def __make_config(self):
         config["worker_nodes"][instance_key] = self.worker_node_type
         config["worker_nodes"][image_key] = self.provider.image
 
+        # NOTE: setup_commands may be list with several sets of shell commands
+        # this change only first set defining the remote environment
+        config["setup_commands"][0] = self._sync_python(
+            config["setup_commands"][0],
+            os.path.join(os.path.dirname(self.__base_config), "setup_commands.sh.in"),
+        )
+
         return _bootstrap_config(config)
 
+    @staticmethod
+    def _sync_python(setup_commands_source: str, setup_commands_template_path: str):
```

Comment:
the whole function looks too specific (expects first line to not change no matter what, deals with one exact package)...
I think we'd better define a function that returns a list of conda package requirements (in this case, `'python={major}.{minor}.{micro}'`) and then have a higher-in-chain function deal with transforming setup section.

This would allow for better syncing other packages in the future, as this separates concerns: one function is knowing _what_ packages to install, and other function knows _how_ to install them.